Kernel/API/POSIX/sys/socket.h

```diff
@@ -9,6 +9,7 @@
 #include <Kernel/API/POSIX/sys/types.h>
 #include <Kernel/API/POSIX/sys/uio.h>
 #include <Kernel/API/POSIX/sys/un.h>
+#include <stddef.h>
```

Comment:
It would be best to not include LibC headers from Kernel/API. See <https://github.com/SerenityOS/serenity/issues/17138>

Reply:
basically, I need `NULL` here. should I move `NULL` into the kernel API? or should I just return `(void*)0`? I can't use nullptr because this is a c header.

Reply:
i used `(struct cmsghdr*)0` for c++ compatibility, but done.